packages/validator-rules/README.md

```diff
@@ -0,0 +1,88 @@
+# AMP-Toolbox Validator Rules
+
+Queries published AMP Validator rules and extracts information about required
+markup and attributes for all AMP formats.
+
+## Usage
+
+Install via:
+
+```
+$ npm install @ampproject/toolbox-validator-rules@canary
+```
+
+### Including the Module
+
+#### ES Module (Browser)
+
+```javascript
+import validatorRules from '@ampproject/toolbox-validator-rules';
+```
+
+#### CommonJs (Node)
+
+```javascript
+const validatorRules = require('@ampproject/toolbox-validator-rules');
+```
+
+### Using the module
+
+```javascript
+  // Loads the validator rules remotely with default options
+  const rules = validatorRules.fetch();
+
+
+  // The raw unprocessed rules
+  console.log(rules.raw);
+
+  // All tags, combined with their respective attribute lists
+  console.log(rules.tags);
+
+  // All extensions
+  console.log(rules.extensions);
+
+  // Get all tag names used in AMP for Email
+  const tags = rules.getTagsForFormat('AMP4EMAIL');
+
+  // Display their names
+  console.log(tags.map(tag => tag.tagName));
+
+  // Get information about an extension
+  const ext = rules.getExtension('amp-carousel', 'AMP4EMAIL');
```

Comment:
Why do you need to provide the format here? This seems counterintuitive.

Reply:
This is just how validator rules are written - there's more than one extension definition with the same name, but different formats.

Reply:
The implementation uses a different order `getExtension(format, extension)`.

Reply:
Ouch, good catch. Fixed.